lib/ansible/plugins/inventory/docker_swarm.py

```diff
@@ -177,10 +184,19 @@ def _populate(self):
                 self.inventory.add_host(self.node_attrs['ID'])
                 self.inventory.add_host(self.node_attrs['ID'], group=self.node_attrs['Spec']['Role'])
                 self.inventory.set_variable(self.node_attrs['ID'], 'ansible_host', self.node_attrs['Status']['Addr'])
+                if self.get_option('host_uri', True):
+                    self.inventory.set_variable(self.node_attrs['ID'], 'ansible_host_uri',
+                                                "tcp://" + self.node_attrs['Status']['Addr'] + ":2376")
                 if self.get_option('verbose_output', True):
                     self.inventory.set_variable(self.node_attrs['ID'], 'docker_swarm_node_attributes', self.node_attrs)
                 if 'ManagerStatus' in self.node_attrs:
                     if self.node_attrs['ManagerStatus'].get('Leader'):
+                        if self.get_option('host_uri', True):
+                            self.inventory.set_variable(self.node_attrs['ID'], 'ansible_host_uri', "tcp://" +
+                                                        parse_address(self.node_attrs['ManagerStatus']['Addr'])[0] +
```

Comment:
Doesn't `self.node_attrs['ManagerStatus']['Addr']` actually contain the correct port (IIRC, don't want to create a swarm to check :) )? If that's the case, why not use it here? Also, why not call `parse_address(self.node_attrs['ManagerStatus']['Addr'])` simply once?

Reply:
It contains the `2377` which is the communication port for swarm nodes port, not the management port.
We can optimize it to just using the `parse_address()` once but it won't give any benefit to a number of code lines, just one less method execution and some memory reserved for a variable. I see no real benefit.

Reply:
Ah, makes sense. I just remembered that there was a port, and somehow assumed it's the one you need here...

Just executing code once is already a real benefit. Saves some cycles. Also, it improves readability since a fitting name will show the reader directly what `parse_address(...)[0]` is :)

Reply:
Ok, will add this with the port detection change as well